transaction/issuance.go

```diff
@@ -125,6 +125,11 @@ func NewTxIssuance(
 	if err != nil {
 		return nil, err
 	}
+
+	if assetAmount == 0 {
+		confAssetAmount = []byte{0}
+	}
+
```

Comment:
instead of adding this here, we could rename the actual `toConfidentialTokenAmount()` to a more generic `toConfidentialIssuanceAmount()`, and replace with it any occurrence of `toConfidentialAssetAmount()` and `toConfidentialTokenAmount()`.